pkg/registry/apis/folders/register.go

```diff
@@ -154,42 +154,57 @@ func (b *FolderAPIBuilder) PostProcessOpenAPI(oas *spec3.OpenAPI) (*spec3.OpenAP
 	return oas, nil
 }
 
+type authorizerParams struct {
+	user identity.Requester
+	eval accesscontrol.Evaluator
+	dec  authorizer.Decision
+	req  string
+}
+
 func (b *FolderAPIBuilder) GetAuthorizer() authorizer.Authorizer {
-	return authorizer.AuthorizerFunc(
-		func(ctx context.Context, attr authorizer.Attributes) (authorized authorizer.Decision, reason string, err error) {
-			verb := attr.GetVerb()
-			name := attr.GetName()
-			if (!attr.IsResourceRequest()) || (name == "" && verb != utils.VerbCreate) {
-				return authorizer.DecisionNoOpinion, "", nil
-			}
-
-			// require a user
-			user, err := identity.GetRequester(ctx)
-			if err != nil {
-				return authorizer.DecisionDeny, "valid user is required", err
-			}
-
-			scope := dashboards.ScopeFoldersProvider.GetResourceScopeUID(name)
-			eval := accesscontrol.EvalPermission(dashboards.ActionFoldersRead, scope)
-
-			// "get" is used for sub-resources with GET http (parents, access, count)
-			switch verb {
-			case utils.VerbCreate:
-				eval = accesscontrol.EvalPermission(dashboards.ActionFoldersCreate)
-			case utils.VerbPatch:
-				fallthrough
-			case utils.VerbUpdate:
-				eval = accesscontrol.EvalPermission(dashboards.ActionFoldersWrite, scope)
-			case utils.VerbDeleteCollection:
-				fallthrough
-			case utils.VerbDelete:
-				eval = accesscontrol.EvalPermission(dashboards.ActionFoldersDelete, scope)
-			}
-
-			ok, err := b.accessControl.Evaluate(ctx, user, eval)
-			if ok {
-				return authorizer.DecisionAllow, "", nil
-			}
-			return authorizer.DecisionDeny, "folder", err
-		})
+	return authorizer.AuthorizerFunc(func(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
+		in := authorizerFunc(ctx, attr)
+
+		if in.user == nil {
+			return in.dec, in.req, nil
+		}
+
+		ok, err := b.accessControl.Evaluate(ctx, in.user, in.eval)
+		if ok {
+			return authorizer.DecisionAllow, "", nil
+		}
+		return authorizer.DecisionDeny, "folder", err
+	})
+}
+
+func authorizerFunc(ctx context.Context, attr authorizer.Attributes) authorizerParams {
+	verb := attr.GetVerb()
+	name := attr.GetName()
+	if (!attr.IsResourceRequest()) || (name == "" && verb != utils.VerbCreate) {
+		return authorizerParams{user: nil, eval: nil, dec: authorizer.DecisionNoOpinion, req: ""}
+	}
+
+	// require a user
+	user, err := identity.GetRequester(ctx)
+	if err != nil {
+		return authorizerParams{user: user, eval: nil, dec: authorizer.DecisionDeny, req: "valid user is required"}
+	}
+
+	scope := dashboards.ScopeFoldersProvider.GetResourceScopeUID(name)
+	eval := accesscontrol.EvalPermission(dashboards.ActionFoldersRead, scope)
```

Comment:
nit - you don't really need to declare this here.  this can be the default return on 209 and you can just return from each switch case rather than changing this eval variable